Source/Initialization/WarpXInitData.cpp

```diff
@@ -1308,8 +1375,10 @@ void WarpX::CheckKnownIssues()
             mypc->m_B_ext_particle_s != "none" || mypc->m_E_ext_particle_s != "none"
         );
         WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
-            !external_particle_field_used,
-            "The hybrid-PIC algorithm does not work with external fields "
+            (!external_particle_field_used
+            || mypc->m_B_ext_particle_s == "parse_b_ext_particle_function"
+            || mypc->m_E_ext_particle_s == "parse_e_ext_particle_function"),
+            "The hybrid-PIC algorithm only works with analytical external E/B fields "
```

Comment:
Is this change still necessary or is it left over from earlier versions of this implementation when it used the particle fields?

```suggestion
            !external_particle_field_used,
            "The hybrid-PIC algorithm does not work with external fields "
```